### PR TITLE
Keep separate commits for PR fixups

### PR DESCRIPTION
**Pull request by tsmaeder**
Just added new instructions for PR's. Please let me know if it's unclear or if you disagree.

### CODE REVIEW

**Review by JonasHelming — APPROVED** [review]

**Review by vince-fugnitto — APPROVED** [review]
The updated ways of working look fine to me :+1:

File: doc/pull-requests.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ Contributors guide: https://github.com/eclipse-theia/theia/blob/master/CONTRIBUT
 <a name="checklist-commit-history"></a>
 - [10.](#checklist-commit-history) Commit history is rebased on master and contains only meaningful commits and changes (less are usually better).
   - For example, use `git pull -r` or `git fetch && git rebase` to pick up changes from the master.
+  - When addressing change requests on a PR, the PR author should provide those changes in separate commits while the review process is not finished. This allows reviewers to re-review only the updated parts of the PR and to determine what needs to be tested again. The "fixup" commits must be squashed before merging in order to keep a clean history.
```
Comment (paul-marechal) [inline doc/pull-requests.md:84]: I would put this higher in the list, under `Opening a Pull Request` so that it's one of the first things mentioned. It not only applies to addressing comments, but any modification could be its own commit to be squashed before merge.

**Review by paul-marechal — APPROVED** [review]

File: doc/pull-requests.md (hunk 1 above)
Comment (tsmaeder) [reply doc/pull-requests.md:84]: Separate commits become important once someone has actually started testing or reviewing the code. So once you've opened a PR, you should push further changes as separate commits. I'll reformulate....

**Review by vince-fugnitto — APPROVED** [review]

**Review by paul-marechal — APPROVED** [review]
